pkg/executor/plan_replayer.go

```diff
@@ -467,6 +468,99 @@ func loadStats(ctx sessionctx.Context, f *zip.File) error {
 	return h.LoadStatsFromJSON(context.Background(), ctx.GetInfoSchema().(infoschema.InfoSchema), jsonTbl, 0)
 }
 
+func unzip(z *zip.File) (string, error) {
+	ra, err := z.Open()
+	if err != nil {
+		return "", err
+	}
+	//nolint: errcheck
+	defer ra.Close()
+	bufa := new(bytes.Buffer)
+	_, err = bufa.ReadFrom(ra)
+	if err != nil {
+		return "", err
+	}
+	return bufa.String(), err
+}
+
+func getTableNameFromZipFilename(name string) string {
+	// dbName.tableName.schema.txt
+	s := strings.Split(name, ".")
+	intest.Assert(len(s) == 4, "invalid zip file name")
+	t := strings.Split(s[0], "/")
+	return t[1] + "." + s[1]
+}
+
+func topoSortTable(input []*zip.File) (result []*zip.File, err error) {
```

Comment:
How about turn off [foreign_key_checks](https://docs.pingcap.com/zh/tidb/stable/system-variables#foreign_key_checks) temporarily in the session? Then it will allow creating child table first :thinking: .

Reply:
Great, I have refactored the code according to your suggestion.